packages/hub/src/graphql/mutations/createRelativeValuesDefinition.ts

```diff
@@ -80,37 +110,21 @@ builder.mutationField("createRelativeValuesDefinition", (t) =>
         type: [RelativeValuesClusterInput],
         required: true,
       }),
+      recommendedUnit: t.input.string({
+        validate: validateItemId,
+      }),
     },
     resolve: async (_, { input }, { session }) => {
-      const email = session?.user.email;
-      if (!email) {
-        // shouldn't happen because we checked user auth scope previously, but helps with type checks
-        throw new Error("Email is missing");
-      }
-
-      if (!input.items.length) {
-        throw new Error(
-          "RelativeValuesDefinition must include at least one item"
-        );
-      }
-
-      const itemIds = new Set<string>();
-      for (const item of input.items) {
-        if (itemIds.has(item.id)) {
-          throw new Error(`Duplicate item id ${item.id}`);
-        }
-        if (!item.id.match(/^\w[\w\-]*$/)) {
-          throw new Error(`Invalid item id ${item.id}`);
-        }
-        itemIds.add(item.id);
-      }
-
-      // TODO - check that `recommendedUnit` matches some item id, and that `clusterId` matches some cluster id
+      validateRelativeValuesDefinition({
+        items: input.items,
+        clusters: input.clusters,
+        recommendedUnit: input.recommendedUnit,
+      });
 
       const definition = await prisma.relativeValuesDefinition.create({
         data: {
           owner: {
-            connect: { email },
+            connect: { email: session.user.email },
```

Comment:
I'm a bit confused here. Are you basically treating the user email as the user ID? (I would have expected you to use User Id, but I'm not used to this code)

Reply:
NextAuth stores the email in the session, but not the user's id, so I'm using it to look up the owner.

But it's still referenced by user id in the database; Prisma's [connect](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#connect-a-single-record) semantics is "look up this related entity and link to it".